packages/web-components/src/drawer/drawer.template.ts

```diff
@@ -0,0 +1,32 @@
+import { ElementViewTemplate, html, ref } from '@microsoft/fast-element';
+import type { Drawer } from './drawer.js';
+import { DrawerType } from './drawer.options.js';
+
+/**
+ * The template for the Drawer component.
+ * @public
+ */
+export function drawerTemplate<T extends Drawer>(): ElementViewTemplate<T> {
+  return html<T>`
+    <dialog
+      class="dialog"
+      part="dialog"
+      role="${x => (x.type === DrawerType.alert ? 'alertdialog' : void 0)}"
+      aria-modal="${x => (x.type === 'non-modal' || x.inline ? void 0 : 'true')}"
+      aria-describedby="${x => x.ariaDescribedby}"
+      aria-labelledby="${x => x.ariaLabelledby}"
+      aria-label="${x => x.ariaLabel}"
+      size="${x => x.size}"
+      position="${x => x.position}"
+      type="${x => x.type}"
+      ?inline="${x => x.inline}"
+      @click="${(x, c) => x.clickHandler(c.event as MouseEvent)}"
+      @keydown="${(x, c) => x.keydownHandler(c.event as KeyboardEvent)}"
```

Comment:
Wrote this below, but you might not need this and can rely on the native `cancel` event. 

Here's what I ended up with on Dialog...

```
@cancel="${(x, c) => (x.type === DialogType.alert ? c.event.preventDefault() : x.hide())}"
```

The difference here is that you maybe never have `alert` drawers and you always want `non-modal` drawers to close on ESC? You may still need to handle the `cancel` event to deal with Safari and emitting events properly.